### PR TITLE
Add parameters to skip snv calling and repeat analysis

### PR DESCRIPTION
**Pull request by ramprasadn**
## PR checklist

Adding two new parameters to make the workflows more modular. I did so we can skip certain workflows for specific analysis. For example, skip repeat calling for wes samples. 

Also includes some minor refactoring to group related workflows together and make them more obvious in the main workflow. 
- [ ] This comment contains a description of changes (with reason).
- [ ] If you've fixed a bug or added code that should be tested, add tests!
- [ ] If you've added a new tool - have you followed the pipeline conventions in the [contribution docs](https://github.com/nf-core/raredisease/tree/master/.github/CONTRIBUTING.md)
- [ ] If necessary, also make a PR on the nf-core/raredisease _branch_ on the [nf-core/test-datasets](https://github.com/nf-core/test-datasets) repository.
- [ ] Make sure your code lints (`nf-core lint`).
- [ ] Ensure the test suite passes (`nextflow run . -profile test,docker --outdir <OUTDIR>`).
- [ ] Ensure the test suite passes (`nextflow run . -profile test_one_sample,docker --outdir <OUTDIR>`).
- [ ] Check for unexpected warnings in debug mode (`nextflow run . -profile debug,test,docker --outdir <OUTDIR>`).
- [ ] Usage Documentation in `docs/usage.md` is updated.
- [ ] Output Documentation in `docs/output.md` is updated.
- [ ] `CHANGELOG.md` is updated.
- [ ] `README.md` is updated (including new tool citations and authors/contributors).

### CODE REVIEW

File: workflows/raredisease.nf [hunk 1]
```diff
@@ -406,63 +416,150 @@ workflow RAREDISEASE {
...
-        )
-        ch_versions = ch_versions.mix(CALL_REPEAT_EXPANSIONS.out.versions)
+/*
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+    CALL AND ANNOTATE REPEAT EXPANSIONS
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+*/
+
+    if (!params.skip_repeat_analysis) {
+        if ( params.analysis_type.equals("wgs") ) {
```
Comment (jemten) [inline workflows/raredisease.nf:426]: Wouldn't it be better to write this in one if statement?
`if (!params.skip_repeat_analysis && params.analysis_type.equals("wgs"))`

File: workflows/raredisease.nf [hunk 2]
```diff
@@ -32,6 +31,14 @@ if (params.run_rtgvcfeval) {
     mandatoryParams += ["rtg_truthvcfs"]
 }
 
+if (!params.skip_repeat_analysis) {
+    mandatoryParams += ["variant_catalog"]
+}
+
+if (!params.skip_snv_calling) {
+    mandatoryParams += ["genome"]
```
Comment (jemten) [inline workflows/raredisease.nf:39]: `genome` is also used by smncopynumbercaller, but maybe you're adding that in a later PR :)

**Review by jemten — APPROVED** [review]
Nice to get it more coherent. I think it would be good to refactor the check for mandatory params in nf-core/raredisease. To me it would be preferable to spin that out to another subworkflow. Might be possible to do it here: 
https://github.com/nf-core/raredisease/blob/dev/subworkflows/local/utils_nfcore_raredisease_pipeline/main.nf 

as it's done in gms/nallo:
https://github.com/genomic-medicine-sweden/nallo/blob/6fa3d442c20581a6bd718bbe09dfe8475a7d7974/subworkflows/local/utils_nfcore_nallo_pipeline/main.nf#L168

separate PR though

File: workflows/raredisease.nf (hunk 2 above)
Comment (ramprasadn) [reply workflows/raredisease.nf:39]: Yupe! I have a couple more PR's in line to be merged after this :D

**Pull request merged**